### PR TITLE
Gulp support

### PR DESCRIPTION
**Pull request by robdodson**
This is a work in progress and not ready to be merged just yet :)

- [x] Ruby SASS processed for app styles
- [x] Ruby SASS processed for element styles
- [x] LibSASS processed for element styles
- [x] Vanilla CSS processed for app styles
- [x] Vanilla CSS processed for element styles
- [x] Verify JSHint is working
- [x] Verify image processing is working
- [x] Verify fonts are being copied properly
- [x] Verify build paths in HTML are being rewritten
- [x] Verify vulcanize works properly for Ruby SASS projects
- [x] Verify vulcanize works properly for LibSASS projects
- [x] Verify vulcanize works properly for CSS projects
- [x] Verify serve works properly for LibSASS projects
- [x] Verify serve works properly for Ruby SASS projects
- [x] Verify serve works properly for CSS projects
- [x] Verify serve:dist works properly for LibSASS projects
- [x] Verify serve:dist works properly for Ruby SASS projects
- [x] Verify serve:dist works properly for CSS projects
- [x] Verify pagespeed is working
- [x] Verify autoprefixr works for CSS
- [x] Verify autoprefixr works for Ruby SCSS
- [x] Verify autoprefixr works for LibSCSS
- [x] Figure out if it's possible to merge the `styles` and `elements` task
- [x] Grunt projects are still generated OK :)

### CODE REVIEW

**Comment by hemanth** [comment]
WOW! Nice list :)

**Comment by robdodson** [comment]
Hah yeah :D

Most of the list I think is already done, I'm just double checking it an making some final decisions. I'd love reviewers when it's wrapped up as I'm still a bit of a gulp newb

**Comment by ragingwind** [comment]
Great. I think I can help you as much as I know.

**Comment by amannn** [comment]
Great stuff! But I'd be careful about LibSASS these days. I recently switched to Ruby SASS, since LibSASS isn't able to compile `/deep/` selectors (see the [issue here](https://github.com/sass/libsass/issues/452)).

**Comment by robdodson** [comment]
@amannn thanks for the heads up on LibSass. I may remove the option for libsass until that bug is fixed.

**Comment by JosefJezek** [comment]
Also CSSO isn't able to compile `/deep/` selectors

https://github.com/css/csso/issues/209

**Comment by robdodson** [comment]
OK, I think I've addressed all the major issues. @addyosmani @sindresorhus if either of you have time to review this before I merge I would be most appreciative. The one thing I was not able to figure out was how to merge the `styles` and `elements` tasks into a single task. Every time I tried, I ended up spraying the files into the same directories :\

File: app/templates/gulpfile.js [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+    ['copy', 'styles'],
+    'elements',
+    ['jshint', 'images', 'fonts', 'html'],
+    'vulcanize',
+    cb);
+});
+
+// Run PageSpeed Insights
+// Update `url` below to the public URL for your site
+gulp.task('pagespeed', pagespeed.bind(null, {
```
Comment (sindresorhus) [inline app/templates/gulpfile.js]: Update to the latest psi and see the updated usage: https://github.com/google/web-starter-kit/blob/ff33b9ab2af1005afbd704326fb484a6af502fee/gulpfile.js#L179

**Comment by sindresorhus** [comment]
Just looking at the diff, lgtm.

**Comment by sindresorhus** [comment]
@JosefJezek We're using clean-css, not CSSO.

File: app/templates/gulpfile.js (hunk 1 above)
Comment (addyosmani) [inline app/templates/gulpfile.js]: You should mostly be able to steal the above code snippet from our changes.

**Comment by addyosmani** [comment]
@JosefJezek In Blink/Chrome, there are conversations on-going about whether or not we're going to keep `/deep/` around at all. Whilst this is needed for Polymer 0.5, I wouldn't gate landing this on that as 0.8 is likely to use it's own theming system based on mixins and props. 

@robdodson We'll review again post the PSI changes.

**Comment by robdodson** [comment]
Closed via 5011f4205c235faf3cc4c7e23435b17653d99862

**Pull request closed without merging**